cli/src/commands/prev.rs

```diff
@@ -78,25 +76,12 @@ pub(crate) fn cmd_prev(
             .heads()
             .contains(current_wc_id);
     let current_wc = workspace_command.repo().store().get_commit(current_wc_id)?;
-    let start_id = if edit {
-        current_wc_id
-    } else {
-        match current_wc.parent_ids() {
-            [parent_id] => parent_id,
-            _ => return Err(user_error("Cannot run `jj prev` on a merge commit")),
-        }
-    };
-    let ancestor_expression = RevsetExpression::commit(start_id.clone()).ancestors_at(offset);
-    let target_revset = if edit {
-        ancestor_expression
-    } else {
-        // Jujutsu will always create a new commit for prev, even where Mercurial cannot
-        // and fails. The decision and all discussion around it are available
-        // here: https://github.com/martinvonz/jj/pull/1200#discussion_r1298623933
-        //
-        // If users ever request erroring out, add `.ancestors()` to the revset below.
-        ancestor_expression.minus(&RevsetExpression::commit(current_wc_id.clone()))
-    };
+    // This is the distance of the target ancestor from the current working copy
+    // revision (@). The offset provided by the user is the distance of the
+    // target ancestor from the parent(s) of the working copy revision (@-).
+    let target_ancestor_distance = if edit { args.offset } else { args.offset + 1 };
```

Comment:
nit: to make it look similar to `cmd_next()`, it might be better to stick to the original structure and turn `start_id` into `Vec<CommitId>` or `RevsetExpression` instead.

Reply:
I pushed a new revision using revset expressions instead of arithmetic on the offset. Please take a look :)